.github/workflows/lint.yml

```diff
@@ -0,0 +1,123 @@
+# Copyright 2020 Vectorized, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+name: Lint all codez
+on:
+  push:
+    tags: 'release-*'
+    branches:
+      - dev
+      - release
+  pull_request:
+    branches:
+      - main
+      - dev
+
+jobs:
+  go:
+    name: Lint go files
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.13.6
+      id: go
+
+    - name: install crlfmt
+      run: |
+        # Workaround to download the specific version we use internally
+        cat > go.mod << EOF
+        module vectorized
+        require (
+          github.com/cockroachdb/crlfmt v0.0.0-20200309141329-97b747f03b21
+        )
+        go 1.13
+
+        EOF
+
+        go mod download
+        go install github.com/cockroachdb/crlfmt
+        rm -rf go.mod
+
+    - name: lint go files
+      run: |
+        find . -name *.go -type f | xargs -n1 crlfmt -wrap=80 -w
+        git diff --exit-code
+
+  js:
+    name: Lint js files
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Set up Node
+      uses: actions/setup-node@v2-beta
+      with:
+        node-version: '12.16.1'
+
+    - name: Set up prettier
+      run: npm install prettier
+
+    - name: Lint JS files
+      run: npx prettier --list-different src/**/*.{ts,js}
```

Comment:
I think there's a script in `package.json` with the desired command.

Reply:
There seem to be many fmt-related ones:
```"scripts": {
    "fmt": "./node_modules/typescript-formatter/bin/tsfmt --baseDir ../../src/js/ -r",
    "build:ts": "npx tsc --build ../../src/js/tsconfig.json",
    "build": "npm run generate:serialization && npm run build:ts",
    "test": "./node_modules/mocha/bin/mocha --recursive \"./output/test\"",
    "prettier:format": "npx prettier --write .",
    "prettier:check": "npx prettier --list-different .",
    "eslint:check": "eslint . --ext .ts",
    "code:ckeck": "npm run prettier:check && npm run eslint:check",
    "generate:serialization": "./generate-entries.sh && npm run prettier:format",
    "postinstall": "npm run generate:serialization",
    "build:public": "./compact-public.js",
    "start": "sudo npx ts-node ./modules/rpc/service.ts"
  }```

@andresaristizabal can you tell us which ones are relevant for the CI pipeline?

Reply:
you can use `npm run prettier:check`

Reply:
Oh good to know. At least I got the command right ;)
That would require me to fetch all of the deps in the package.json though, right?

Also, I'd like to keep this as close to our internal build ( which runs the bare command). It would be good to change the internal one to use the scripts from package.json though.